descartes_light/include/descartes_light/interface/edge_evaluator.h

```diff
@@ -32,13 +32,59 @@ template <typename FloatType>
 class EdgeEvaluator
 {
 public:
+  using Ptr = typename std::shared_ptr<EdgeEvaluator<FloatType>>;
+  EdgeEvaluator(std::size_t dof) : dof_(dof) {}
   virtual ~EdgeEvaluator() {}
 
   virtual bool evaluate(const Rung_<FloatType>& from,
```

Comment:
I don't really like the function signature here. Why does this function return false if the two rungs aren't connected by any vertices? It's not like the evaluation itself failed. I would propose returning the vector of edge lists instead; then let the user decide what it means to have a vector of edge lists that is empty.

```c++
std::vector<typename LadderGraph<FloatType>::EdgeList> evaluate(const Rung_<FloatType>& from, const Rung_<FloatType>& to)
```

Reply:
> Why does this function return false if the two rungs aren't connected by any vertices? It's not like the evaluation itself failed. I would propose returning the vector of edge lists instead; then let the user decide what it means to have a vector of edge lists that is empty.

This is an interface class that is only to used by the descartes code which keeps the building of the graph cleaner. The user should not directly be calling evaluate or considerEdge.

Reply:
In this case, the "user" I'm referring to is actually us as developers, not necessarily end-users. I think a change to this interface would clear up the ambiguity with `edges` resizing we're discussing [above](https://github.com/swri-robotics/descartes_light/pull/26#discussion_r499662078). It would be really easy to create a helper function that can look through the output list of edges and determine if it is empty or not to replace the boolean check [here](https://github.com/swri-robotics/descartes_light/blob/e3f6d85063fea4ac77305ea4c180e7c151c1c374/descartes_light/include/descartes_light/impl/descartes_light.hpp#L118). This wouldn't introduce much additional code to the graph construction and would improve the clarity of the edge evaluator interface

Reply:
Ok, make sense. Lets create an issue on this and make the change in a separate PR. You good with this?